### PR TITLE
Changes to Override the AMQP BasicProperties

### PR DESCRIPTION
**Pull request by vishnu-alapati**
### Description of the Change

Made Changes to Override the AMQP BasicProperties, earlier it was using wrong contentType while sending the Events

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Vishnu Alapati vishnu.alapati@tcs.com

### CODE REVIEW

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java [hunk 1]
```diff
@@ -492,8 +494,13 @@ public void shutdownCompleted(ShutdownSignalException cause) {
                 }
             });
             BasicProperties msgProps = MessageProperties.BASIC;
+            final BasicProperties PERSISTENT_BASIC_APPLICAION =
```
Comment (z-sztrom) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java:497]: Initialization of PERSISTENT_BASIC_APPLICATION should be stored in MessageProperties class, next to intialization of other properties, e.g. MINIMAL_BASIC, BASIC, etc. Initialization done here causes that a new instance of the BasicProperties is created every time send() method is invoked.

Comment (z-sztrom) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java:497]: I'd follow convetions given by MessageProperties and rename PERSISTENT_BASIC_APPLICATION -> PERSISTENT_BASIC_JSON or PERSISTENT_BASIC_APPLICATION_JSON.

Comment (z-sztrom) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java:497]: I didn't notice the MessageProperties class is part of rabbitmq client library:-( Initialization of the properties should be done at class level of RabbitMqProperties.

Comment (vishnu-alapati) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java]: Made the proper changes

Comment (vishnu-alapati) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java]: Renamed it

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,325 +1,328 @@
...
-- added optional authentication to an Active Directory server for all 
-  REST endpoints
-
-
-## 0.1.8
-- Improved error handling/logging
+##2.0.28
+- Added the code changes to override the AMQP.BasicProprties.
+
+## 2.0.27
```
Comment (z-sztrom) [inline CHANGELOG.md:4]: Have you changed newlines or something else in the file? Diff shows the file is completely different...

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java [hunk 3]
```diff
@@ -493,7 +503,7 @@ public void shutdownCompleted(ShutdownSignalException cause) {
             });
             BasicProperties msgProps = MessageProperties.BASIC;
```
Comment (z-sztrom) [inline publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/helper/RabbitMqProperties.java:504]: Why always assign BASIC properties to msgProps and then overwrite it? I'd do

`BasicProperties msgProps = usePersistence ? PERSISTENT_BASIC_APPLICATION_JSON : MessageProperties.BASIC;`

**Pull request closed without merging**